internal/app/secretless/handlers/mysql/handler.go

```diff
@@ -2,58 +2,97 @@ package mysql
 
 import (
 	"fmt"
+	"github.com/cyberark/secretless-broker/internal/app/secretless/handlers/mysql/protocol"
+	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
 	"io"
 	"log"
 	"net"
+	"reflect"
 	"strconv"
-
-	"github.com/cyberark/secretless-broker/internal/app/secretless/handlers/mysql/protocol"
-	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
 )
 
-// BackendConfig stores the connection info to the real backend database.
-// These values are pulled from the handler credentials config
-type BackendConfig struct {
-	Host     string
-	Port     uint
-	Username string
-	Password string
-	Options  map[string]string
-}
-
-func (bc *BackendConfig) Address() string {
-	return bc.Host + ":" + strconv.FormatUint(uint64(bc.Port), 10)
-}
-
 // Handler connects a client to a backend. It uses the handler Config and Providers to
-// establish the BackendConfig, which is used to make the Backend connection. Then the data
+// establish the ConnectionDetails, which is used to make the Backend connection. Then the data
 // is transferred bidirectionally between the Client and Backend.
 //
 // Handler requires "host", "port", "username" and "password" credentials.
+//
 type Handler struct {
-	// SequenceIds keep track of packet no. for client and backend
-	// MySQL uses this to ensure packets arrive in order
-	// values indicate appropriate value to use on subsequent write
-	clientSequenceId byte
-	backendSequenceId byte
+
+	// The connections are decorated versions of net.Conn that allow us
+	// to do read/writes according to the MySQL protocol.  Protocol level
+	// details are thus encapsulated.  Within the MySQL code, we _only_
+	// deal with these decorated versions.
+
+	mySQLClientConn *Connection
+	mySQLBackendConn *Connection
 	plugin_v1.BaseHandler
-	BackendConfig *BackendConfig
+	connectionDetails *ConnectionDetails
 }
 
-func (h *Handler) abort(err error) {
```

Comment:
my sense is that we're dropping this because we've implemented even better error handling - is this fair? I know we were aborting the connection attempt in certain circumstances before

Reply:
yes this has essentially been moved around
the logic still exists to abort